### PR TITLE
Add short stitch feature to satin column

### LINKED ISSUE #308: Satin: short stitches in corners
On the inside of a sharp corner, stitches can pile up.  We can reduce the density by having some of the stitches fall short of the corner.  Ideally ink/stitch would do this automatically, perhaps with an option to disable.

### PR DESCRIPTION
**Pull request by kaalleen**
Fixes #308

Since this feature is supposed to be a satin column default, I will need help to get the defaults right.

### CODE REVIEW

**Comment by lexelby** [comment]
I haven't tested this out yet, but... I think this will just make all the stitches in a section short, right?  As I understand it, other programs only make _some_ of the stitches short, but keep some the full length.  That way the satin still has the right shape.

**Comment by kaalleen** [comment]
Every second stitch in sections with a defined density.

File: lib/elements/satin_column.py [hunk 1]
```diff
@@ -99,6 +99,24 @@ def e_stitch(self):
...
+        return self.get_float_param("short_stitch_inset", 10) / 100
+
+    @property
+    @param('short_stitch_distance',
+           _('Short stitch distance'),
+           tooltip=_('Do short stitches if the distance between stitches is smaller than this.'),
+           type='float', unit="mm",
+           default=0.25)
+    def short_stitch_distance(self):
+        return self.get_float_param("short_stitch_distance", 0.25) * PIXELS_PER_MM
```
Comment (lexelby) [inline lib/elements/satin_column.py:118]: If you name the param `short_stitch_distance_mm` (on this line and on 112) then you don't have to multiply by PIXELS_PER_MM.  `get_float_param()` will do it for you.  You can still use `def short_stitch_distance(self):`, you don't have to rename the function too.

File: lib/elements/satin_column.py [hunk 2]
```diff
@@ -898,6 +914,21 @@ def _get_split_points(self, left, right, max_stitch_length, count=None):
...
+            if left.distance(sides[0][i-1]) < self.short_stitch_distance:
+                split_point = self._get_split_point(left, right, self.short_stitch_inset)
+                sides[0][i] = Point(split_point.x, split_point.y)
+            if right.distance(sides[1][i-1]) < self.short_stitch_distance:
+                split_point = self._get_split_point(right, left, self.short_stitch_inset)
+                sides[1][i] = Point(split_point.x, split_point.y)
+
+    def _get_split_point(self, point1, point2, distance_fraction):
+        line = shgeo.LineString([point1, point2])
+        return line.interpolate(distance_fraction, normalized=True)
```
Comment (lexelby) [inline lib/elements/satin_column.py:930]: Cool trick:

`return point1 * (1 - distance_fraction) + point2 * distance_fraction`

Comment (lexelby) [reply lib/elements/satin_column.py:930]: And as an added bonus, that returns a Point object.

**Comment by lexelby** [comment]
Oh, I missed that!  That's perfect!  I added a few comments but this looks great.

**Comment by kaalleen** [comment]
Thank you, I updated according to your comments. That has been helpful :)

**Pull request merged**